test/vm_tests/transformer/impl_generation_test.dart

```diff
@@ -328,6 +328,63 @@ main() {
           expect(transformedSource, contains(transformedLine));
         });
 
+        test('with Props|State with clause contain \$ prefixed and non-prefixed mixin pairs', () {
+          final prefixedFooPropsMixin = 'abstract class \$FooPropsMixin {}';
+          final prefixedBarPropsMixin = 'abstract class \$BarPropsMixin {}';
+          final prefixedFooStateMixin = 'abstract class \$FooStateMixin {}';
+
+          preservedLineNumbersTest('''
+            @Factory()
+            UiFactory<FooProps> Foo;
+        
+            @Props()
+            class FooProps extends UiProps with FooPropsMixin,
+            // TODO: AF-#### This will be removed once the transition to Dart 2 is complete.
+            // ignore: mixin_of_non_class,undefined_class
+            \$FooPropsMixin, BarPropsMixin, \$BarPropsMixin {}
+
+            @State()
+            class FooState extends UiState with FooStateMixin,
+            // TODO: AF-#### This will be removed once the transition to Dart 2 is complete.
+            // ignore: mixin_of_non_class,undefined_class
+            \$FooStateMixin {}
```

Comment:
Since the implementation generation is no longer relying on props or state classes that mixin specific props or state mixins, I think we can ditch this boilerplate just to slim down and simplify the test a bit. Really all we need to test is that all instances of props mixins and state mixins have their `$`-prefixed versions generated, so a test with 2 props mixins and 2 state mixins in the same file should be sufficient

Reply:
In other words, you can ditch the component, factory, props, and state pieces and then add another state mixin